LANGUAGE.md

```diff
@@ -0,0 +1,155 @@
+# Powdr Assembly (Powdr-Asm)
+
+Powdr-Asm is used to define instruction sets and write programs that depend on
+control flow, with the aim of generating ZKPs about them.
+Powdr-Asm follows the execution trace table model shared by many proof systems.
+In fact, Powdr-Asm extends Polygon Hermez' [PIL and
+zkASM](https://zkevm.polygon.technology/docs/category/polynomial-identity-language).
+In such model, the execution is represented by a two-dimensional fixed-size
+table, where each column represents a piece of state (for example registers or
+memory) and each row represents a step in time for a given execution.
+The program is represented by a transition system that specifies how the state
+evolves in time in the form of constraints.
+
+As an example, let's apply that model conceptually to a program that computes
+the factorial of a number. Our state consists of two registers:
+
+- CNT - a counter
+- ACC - the answer accumulator
+
+The first row will represent the initial state of our program, that is, `PC =
+0`, `CNT = <input>`, `ACC = 1`.
+From then on, we compute the subsequent rows with the following pseudocode:
+
+1. START:
```

Comment:
I think numbering the program lines confuses because people could think that the program lines correspond to the table rows.